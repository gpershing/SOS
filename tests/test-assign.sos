//int
a: int = 5
print(a) //5
a = 6
print(a) //6


//float
b: float = 1.0
printf(b) //1.0
b = 2.5
printf(b) //2.5


//bool
c: bool = true
print(c) //True
c = false
print(c) //False


//array
d: array int = [1, 2]
print(d[0]) //1
temp: int = d[1]
d[1] = d[0]
d[0] = temp
print(d[0]) //2


//struct
struct point = {x: float, y: float}
struct e = {v: int, w: float, x: bool, y: point, z: array int }

es: e = {4, 4.4, true, {3.3, 4.5}, [5, 6, 7]}
print(es.v) //4
printf(es.w) //4.4
print(es.x) //True
printf(es.y.x) //3.3
print(es.z[0]) //5
