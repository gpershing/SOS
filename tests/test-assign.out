5
6
1
2.5
1
0
1
2
4
4.4
1
3.3
5
